hydra-node/src/Hydra/HeadLogic/Outcome.hs

```diff
@@ -84,22 +113,45 @@ data StateChanged tx
       , newLocalUTxO :: UTxOType tx
       , newLocalTxs :: [tx]
       }
-  | CommitFinalized {newVersion :: SnapshotVersion, depositTxId :: TxIdType tx}
-  | DecommitFinalized {newVersion :: SnapshotVersion}
   | PartySignedSnapshot {snapshot :: Snapshot tx, party :: Party, signature :: Signature (Snapshot tx)}
-  | SnapshotConfirmed {snapshot :: Snapshot tx, signatures :: MultiSignature (Snapshot tx)}
-  | HeadClosed {chainState :: ChainStateType tx, contestationDeadline :: UTCTime}
-  | HeadContested {chainState :: ChainStateType tx, contestationDeadline :: UTCTime}
-  | HeadIsReadyToFanout
-  | HeadFannedOut {chainState :: ChainStateType tx}
-  | ChainRolledBack {chainState :: ChainStateType tx}
+  | SnapshotConfirmed {headId :: HeadId, snapshot :: Snapshot tx, signatures :: MultiSignature (Snapshot tx)}
+  | HeadClosed {headId :: HeadId, snapshotNumber :: SnapshotNumber, chainState :: ChainStateType tx, contestationDeadline :: UTCTime}
+  | HeadContested {headId :: HeadId, snapshotNumber :: SnapshotNumber, chainState :: ChainStateType tx, contestationDeadline :: UTCTime}
+  | HeadIsReadyToFanout {headId :: HeadId}
+  | HeadFannedOut {headId :: HeadId, utxo :: UTxOType tx, chainState :: ChainStateType tx}
+  | -- More constructors to match the mapping needed for the 'ServerOutput' type.
+    ChainRolledBack {chainState :: ChainStateType tx}
   | TickObserved {chainSlot :: ChainSlot}
+  | PeerHandshakeFailure
+      { remoteHost :: Host
+      , ourVersion :: Natural
+      , theirVersions :: [Natural]
+      }
+  | IgnoredHeadInitializing
+      { headId :: HeadId
+      , contestationPeriod :: ContestationPeriod
+      , parties :: [Party]
+      , participants :: [OnChainId]
+      }
+  | PostTxOnChainFailed {postChainTx :: PostChainTx tx, postTxError :: PostTxError tx}
+  | PeerConnected {peer :: NodeId}
+  | PeerDisconnected {peer :: NodeId}
+  | CommandFailed {clientInput :: ClientInput tx, state :: HeadState tx}
+  | GetUTxOResponse {headId :: HeadId, utxo :: UTxOType tx}
+  | InvalidInput {reason :: String, input :: Text}
+  | Greetings
```

Comment:
It feels misleading for these types to be considered StateChanged, as it seems to contradict this [ADR](https://hydra.family/head-protocol/adr/24).

IMO, a 1:1 mapping here isn’t feasible nor desirable, since ServerOutputs also represent responses from the node, not just state changes.